### PR TITLE
Implement logging in via webengine

### PR DESCRIPTION
**Pull request by aznd**
Before, the user would need to manually copy the SID. That wasn't the cleanest solution.
Now, Rare checks if webengine is installed. If yes, it will open a browser where the user can log in.
After logging in (in the browser) it will log in normally. The user won't have to do anything extra.

### CODE REVIEW

**Comment by aznd** [comment]
Currently, the window that gets opened on login, doesn't get the focus.
Neither the user can change the focus. This needs more inspection.

File: rare/components/dialogs/login/browser_login.py [hunk 1]
```diff
@@ -77,4 +78,18 @@ def do_login(self):
...
+            if webview_login.do_webview_login(
+                    callback_sid=self.core.auth_sid,
+                    callback_code=self.core.auth_code):
+                logger.info(
+                    "Successfully logged in as "
+                    f"{self.core.lgd.userdata['displayName']}"
+                )
+                self.success.emit()
+            else:
+                logger.warning("Failed to login. Did you cancel the login?")
```
Comment (Dummerle) [inline rare/components/dialogs/login/browser_login.py:95]: Questions in logging messages are useless

File: optional_requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+pywebview[gtk]
```
Comment (Dummerle) [inline optional_requirements.txt:1]: I like the new optional requirements file. Please add it in setup.py too. See https://github.com/derrod/legendary/blob/cffb10188a11475f417cad394ceb929af44e6773/setup.py#L42 as example. Then you can execute pip install Rare[webview]

**Review by Dummerle — COMMENTED** [review]
I will test it later. But the code looks good

**Pull request merged**